dagshub/data_engine/client/loaders/base.py

```diff
@@ -81,7 +81,22 @@ def _get_file_columns(self):
             [self.entries[0].metadata[col] for col in self.metadata_columns],
         ):
             try:
-                self.repo.list_path((self.datasource_root / str(value)).as_posix())
+                if self.source == "repo":
+                    self.repo.list_path((self.datasource_root / str(value)).as_posix())
```

Comment:
Replace the `self.source` in this class to use the `self.datasource.source.source_type` enum instead of splitting the path. You can compare against the enum then